.github/workflows/pr.yml

```diff
@@ -31,6 +65,19 @@ jobs:
       - name: Setup CI
         uses: ./.github/actions/setup
 
+      - name: Install Athena toolchain
+        run: |
+          curl -L https://install.athenavm.org | bash
+          $ATHENA_DIR/bin/athup --path $GITHUB_WORKSPACE
+          $ATHENA_DIR/bin/cargo-athena athena --version
+          echo "$ATHENA_DIR/bin" >> $GITHUB_PATH
+
+      - name: Install Athena CLI
+        run: |
+          cd cli
+          cargo install --force --locked --path .
+          cargo athena install-toolchain
+
       - name: Run cargo check
```

Comment:
Running `cargo check` here actually rebuilds the binaries: https://github.com/athenavm/athena/actions/runs/11391768813/job/31696289475?pr=159

Maybe the `build-binaries` job isn't necessary?

Reply:
I added build binaries job to avoid installing athena in every job, where it is normally not needed otherwise. I takes quite a time to install athena and build all binaries so it also speeds up CI.

Reply:
I'm confused. This job also includes "Install Athena toolchain" and "Install Athena CLI". How is this saving time, the way you have this set up? Installing the toolchain is relatively slow since it's a big file, but, to reiterate, running `cargo check` here is going to rebuild the binaries _anyway._

Reply:
Yes, `cargo check` will rebuild binaries in this job, but this job specifically does not use the artifacts from `build-binaries`. Its artifacts are used in `test-go` and `lint` jobs.